src/Filters/filt_order.jl

```diff
@@ -0,0 +1,458 @@
+
+        Integer and natural frequency order estimation for $($filt) Filters. `Wp`
+        and `Ws` indicate the passband and stopband frequency edges, and `Rp` and `Rs` indicate
+        the maximum loss in the passband and the minimum attenuation in the stopband, (in dB.)
+        Based on the ordering of passband and stopband edges, the Lowpass or Highpass filter type is inferred.
+        `N` indicates the smallest integer filter order that achieves the desired specifications,
+        and `ωn` contains the natural frequency of the filter, (in this case, simply the passband edge.)
+        If a domain of `:s` is specified, the passband and stopband edges are interpretted
+        as radians/second, giving an order and natural frequency result for an analog filter.
+        The default domain is `:z`, interpretting the input frequencies as normalized from 0 to 1,
+        where 1 corresponds to π radians/sample.
+        """
+        function $fcn(Wp::Real, Ws::Real, Rp::Real, Rs::Real; domain::Symbol=:z)
+            ftype = (Wp < Ws) ? Lowpass : Highpass
+            if (domain == :z)
+                Ωp = tan(π/2 * Wp)
+                Ωs = tan(π/2 * Ws) 
+            else
+                Ωp = Wp
+                Ωs = Ws
+            end
+            # Lowpass/Highpass prototype xform is same as Butterworth.
+            wa = toprototype(Ωp, Ωs, ftype)
+            N = ceil(Int, $(Symbol(string(est, "_order_estimate")))(Rp, Rs, wa))
+            if (domain == :z)
+                ωn = (2/π)*atan(Ωp)
+            else
+                ωn = Ωp
+            end
+            N, ωn
+        end
+
+        """
+            (N, ωn) = $($fcn)(Wp::Tuple{Real, Real}, Ws::Tuple{Real, Real}, Rp::Real, Rs::Real; domain::Symbol=:z)
+
+        Integer and natural frequency order estimation for $($filt) Filters. `Wp` and `Ws` are 2-element
+        frequency edges for Bandpass/Bandstop cases, with `Rp` and `Rs` representing the ripple maximum loss
+        in the passband and minimum ripple attenuation in the stopband in dB. Based on the ordering of passband 
+        and bandstop edges, the Bandstop or Bandpass filter type is inferred. `N` is an integer indicating the 
+        lowest estimated filter order, with `ωn` specifying the cutoff or "-3 dB" frequencies. If a domain of 
+        `:s` is specified, the passband and stopband frequencies are interpretted as radians/second, giving an 
+        order and natural frequencies for an analog filter. The default domain is `:z`, interpretting the input 
+        frequencies as normalized from 0 to 1, where 1 corresponds to π radians/sample.
+        """
+        function $fcn(Wp::Tuple{Real, Real}, Ws::Tuple{Real, Real}, Rp::Real, Rs::Real; domain::Symbol=:z)
+            Wps = (Wp[1] > Wp[2]) ? tuple(Wp[2], Wp[1]) : tuple(Wp[1], Wp[2])
+            Wss = (Ws[1] > Ws[2]) ? tuple(Ws[2], Ws[1]) : tuple(Ws[1], Ws[2])
+            ftype = (Wps[1] < Wss[1]) ? Bandstop : Bandpass
```

Comment:
Like above: Should we verify that `(Wps[1] < Wss[1]) == (Wps[2] > Wss[2])` and throw an informative error otherwise?

Reply:
Added!